### ISSUE TITLE
Clarify SSL support

### ISSUE THREAD

**Issue by ericmj** [fastly/terraform-provider-fastly#29]
According to this comment [1] use_ssl is not supported yet. Does this mean there is no way to connect to the backend over SSL? If so why are there other SSL options?

[1] https://github.com/terraform-providers/terraform-provider-fastly/blob/7801970d12ab6d2b672e54c2ca3986db5d72cb6c/fastly/resource_fastly_service_v1.go#L292-L300

**Comment by ericmj** [comment]
If you can clarify what work needs to be done I'd be glad to contribute it. Right now I can't use this because my backend requires HTTPS and SNI.

**Comment by apparentlymart** [comment]
Hi @ericmj,

From looking at the commit history here it looks like the initial work on this provider added this commented-out `use_ssl` attribute, and then subsequent commits have gradually added many of the other TLS-related settings from [the backend docs](https://docs.fastly.com/api/config#backend), so we may now have enough to re-enable this setting.

It looks like `min_tls_version` and `max_tls_version` are still absent, but I'm not sure if they are absolutely needed to be able to turn on SSL here with this option.

If you have the time and motivation to try it, I'd start by un-commenting this setting and seeing if it works as-is and allows you to produce a viable service. If so, we can take a pretty simple PR to uncomment this, add a test for it, and include it in the docs. (I think we found that while the API would let us turn this option _on_, the result was originally a broken service due to certain other attributes being mandatory in that case. So it'd be important to test that the created service is actually usable.)

If it _doesn't_ work, hopefully testing it will shed light on what work is remaining before it _can_ be used, in which case it'd be nice to document that here and, again if you're willing, possibly add those additional features required to get this over the line.

Any work you're able to do here would be much appreciated, @ericmj! Thanks for pointing this out.

**Comment by ericmj** [comment]
Thank you for the quick response. I will start investigating what works already.

**Comment by msuterski** [comment]
We're using backends over TLS without that option. All the `.ssl_*` options that are supported by TF work as expected for us
```
    ssl_check_cert        = "true"
    ssl_cert_hostname     = "<hostname>"
    ssl_sni_hostname      = "<hostname>"
```

The `use_ssl` one seems not to affect the backend at all. Even for the backends that we define in our VCLs, the `.ssl` option does not seem to do anything. Fastly's support was unable to clearly answer questions about it, too.

**Comment by ericmj** [comment]
@msuterski For me those settings do not show up in the terraform state and they don't seem to affect anything on Fastly when I look in the Fastly UI.

**Comment by msuterski** [comment]
@ericmj which version of TF are you using? We're on 0.9.4. I can see those options in our state files and our backends would fail without them in Fastly configs.

**Comment by ericmj** [comment]
terraform 0.10.0 and terraform-provider-fastly 0.1.2.

**Comment by ericmj** [comment]
Maybe it's rather a state storage issue, see: https://github.com/terraform-providers/terraform-provider-fastly/issues/31#issuecomment-321709555.

**Comment by msuterski** [comment]
try the 0.9.x version with the build in providers.

**Comment by hynd** [comment]
Closed by #40?

**Comment by philippschulte** [comment]
The provider supports all ssl/tls `backend` attributes of the [Fastly API](https://docs.fastly.com/api/config#api-section-backend). Therefore, I'll close this one out!
